nutrition_labels/grant_tagger.py

```diff
@@ -290,29 +297,28 @@ def train_test_info(self):
         }
         """
 
-        y_predict = self.model.predict(self.X_vect)
-        y_predict_proba = self.model.predict_proba(self.X_vect)
-        # The probability of the predicted binary value
-        y_predict_proba = np.max(y_predict_proba, axis=1)
+        X_test_vect = self.vectorizer.transform(X_test)
+        X_unused_vect = self.vectorizer.transform(self.X_unused)
```

Comment:
`X_unused` is not passed in, where did the class find it? can it be the case that this method is passed without `X_unused` being defined? also X_unused sounds like a heavy object to keep in memory and save with the model, is it necessary?

Reply:
it is created when X_test and X_train are created in `split_data`. `train_test_info` is only really designed to be useful to run after training (and therefore `self.X_unused` should exist). But I see your point and maybe `split_data` should return `return X_train, X_test, X_unused, y_train, y_test, y_unused` - a bit unfortunate that it's not in the style of sklearn split though.

Reply:
I think that would be better and there is no way to conform to the sklearn api for all their functions.